jplotter/src/main/java/hageldave/jplotter/util/Utils.java

```diff
@@ -435,4 +438,92 @@ public static List<Field> getReflectionFields(Class<?> toSearch) {
 		}
 		return toFill;
 	}
+
+	/**
+	 * Clips the given line to the rectangle.
+	 *
+	 * @param rect {@link Rectangle2D} object where the line should be clipped to
+	 * @param line {@link Line2D} object to clip
+	 * @return the clipped line
+	 */
+	public static Line2D getClippedLine(Rectangle2D rect, Line2D line) {
+		Point2D intersectionPoint;
+		double x1 = line.getX1(), y1 = line.getY1(), x2 = line.getX2(), y2 = line.getY2();
+
+		Line2D.Double unclippedLine = new Line2D.Double(new Point2D.Double(x1, y1), new Point2D.Double(x2, y2));
+		Line2D.Double leftBorder = new Line2D.Double(new Point2D.Double(rect.getMinX(), rect.getMinY()), new Point2D.Double(rect.getMinX(), rect.getMaxY()));
+		Line2D.Double rightBorder = new Line2D.Double(new Point2D.Double(rect.getMaxX(), rect.getMinY()), new Point2D.Double(rect.getMaxX(), rect.getMaxY()));
+		Line2D.Double topBorder = new Line2D.Double(new Point2D.Double(rect.getMinX(), rect.getMinY()), new Point2D.Double(rect.getMaxX(), rect.getMinY()));
+		Line2D.Double bottomBorder = new Line2D.Double(new Point2D.Double(rect.getMinX(), rect.getMaxY()), new Point2D.Double(rect.getMaxX(), rect.getMaxY()));
+
+		switch (rect.outcode(x1, y1)) {
```

Comment:
switch is probably not a good idea here, because `outcode(...)` can also return logical OR combinations, for example `(OUT_LEFT  | OUT_TOP)`. I think to cover all cases it would be better to check each condition separately by checking if the outcode contains the respective bitmask:
```java
if( (outcode & OUT_TOP)==OUT_TOP){ /* ... intersect with top line, update x,y */}
if( (outcode & OUT_BOTTOM)==OUT_BOTTOM){ /* ... intersect with bottom line, update x,y */}
if( (outcode & OUT_LEFT)==OUT_LEFT){ /* ... intersect with left line, update x,y */}
if( (outcode & OUT_RIGHT)==OUT_RIGHT){ /* ... intersect with right line, update x,y */}

```

Reply:
Yep, sounds reasonable. I replaced the switch cases with if conditions 👍